src/connectors/CustomNetworkConnector.ts

```diff
@@ -130,8 +130,9 @@ export class CustomNetworkConnector extends AbstractConnector {
   }
 
   public switchUnsupportedNetwork(networkDetails?: NetworkDetails) {
-    if (!window.ethereum || !window.ethereum.request || !window.ethereum.isMetaMask || !networkDetails) return
-    window.ethereum
+    if (!window.ethereum || !window.ethereum.request || !window.ethereum.isMetaMask || !networkDetails)
```

Comment:
`!window.ethereum || !window.ethereum.request || !window.ethereum.isMetaMask || !networkDetails`

I think this condition could be a variable

Reply:
Computer says no. :(
Just below your comment there is `window.request` that complains it can be undefined if we have that logic generalised as you are suggesting.
Other than that I would love to be able to implement it.